packages/react-component-library/src/components/Timeline/TimelineRows.tsx

```diff
@@ -81,39 +81,65 @@ export const TimelineRows: React.FC<TimelineRowsProps> = ({
   return (
     <main className={mainClasses}>
       {hasChildren && (
-        <div className={rowClasses} data-testid="timeline-columns">
-          <TimelineContext.Consumer>
-            {({
-              state: {
-                months,
-                weeks,
-                options: { dayWidth },
-              },
-            }) => {
-              return weeks.map(({ startDate }, index) => {
-                const diff = differenceInDays(
-                  new Date(startDate),
-                  new Date(months[0].startDate)
-                )
-
-                const offsetPx = formatPx(
-                  dayWidth,
-                  diff < 0 ? -Math.abs(diff) : 0
-                )
-
-                const widthPx = formatPx(dayWidth, 7)
-
-                const isOddNumber = isOdd(index)
-
-                const column = renderColumns
-                  ? renderColumns(index, isOddNumber, offsetPx, widthPx)
-                  : renderDefaultColumns(index, isOddNumber, offsetPx, widthPx)
-
-                return withKey(column, 'timeline-column', startDate.toString())
-              })
-            }}
-          </TimelineContext.Consumer>
-        </div>
+        <TimelineContext.Consumer>
+          {({
+            state: {
+              months,
+              weeks,
+              endDate: timelineEndDate,
+              options: { dayWidth },
+            },
+          }) => {
+            const wrapperStyles = timelineEndDate
```

Comment:
Rather than this ternary operator here I would find it easier to read if there was a function:

```javascript
  function getOffset(firstMonthStartDate: Date) {
    if (endDate) {
      return formatPx(
        dayWidth,
        differenceInDays(firstMonthStartDate, startDate)
      )
    }

    return null
  }
```

And then `marginLeft: getOffset(months[0].startDate)`

The result is a lot less indentation.

Reply:
The function would actually be more nesting than the ternary as there is another style that needs including:

```typescript
function getOffsetStyles(firstMonthStartDate: Date, timelineEndDate: Date) {
  if (timelineEndDate) {
    return {
      width: formatPx(
        dayWidth,
        differenceInDays(timelineEndDate, firstMonthStartDate) + 1
      ),
      overflow: 'hidden',                  
    }
  }

  return null
}
```

vs

```typescript
const wrapperStyles = timelineEndDate
? {
    width: formatPx(
      dayWidth,
      differenceInDays(timelineEndDate, months[0].startDate) + 1
    ),
    overflow: 'hidden',
  }
: null
```

Reply:
Apologies, I was supposed to add this comment to `Timeline.tsx`

Reply:
We have agreed to leave the ternaries for the dynamic styles and instead push ahead with the upcoming styled-components works for Timeline.